nest-forms-backend/src/convert-pdf/convert-pdf.service.spec.ts

```diff
@@ -65,6 +67,14 @@ describe('ConvertPdfService', () => {
     const module: TestingModule = await Test.createTestingModule({
       providers: [
         FormsHelper,
+        {
+          provide: ConfigService,
+          useValue: createMock<ConfigService>(),
+        },
+        {
+          provide: FormValidatorRegistryService,
+          useValue: createMock<FormValidatorRegistryService>(),
+        },
```

Comment:
config service is provided below - let me know if this is solving some issue, if it is it may be solved by some other reordering ?

note in general - it's better to do this (mocking) on module level. Solves much of the issues with ordering and missing service mocks, makes generally more maintainable and is likely the way it is intended to be done. We did not know of this always. Fine to keep as is in this PR, noting for future and in case you'd encounter future issues with services missing in test mocks.

Reply:
Removed the double provider